src/custom/hooks/useApproveCallback/useApproveCallbackMod.ts

```diff
@@ -24,33 +24,65 @@ export enum ApprovalState {
   APPROVED = 'APPROVED',
 }
 
-// returns a variable indicating the state of the approval and a function which approves if necessary or early returns
-export function useApproveCallback(
-  openTransactionConfirmationModal: (message: string) => void,
-  closeModals: () => void,
-  amountToApprove?: CurrencyAmount<Currency>,
+export interface ApproveCallbackParams {
+  openTransactionConfirmationModal: (message: string) => void
+  closeModals: () => void
+  amountToApprove?: CurrencyAmount<Currency>
   spender?: string
-): [ApprovalState, (optionalParams?: OptionalApproveCallbackParams) => Promise<void>] {
+  amountToCheckAgainstAllowance?: CurrencyAmount<Currency>
+}
+
+// returns a variable indicating the state of the approval and a function which approves if necessary or early returns
+export function useApproveCallback({
+  openTransactionConfirmationModal,
+  closeModals,
+  amountToApprove,
+  spender,
+  amountToCheckAgainstAllowance,
+}: ApproveCallbackParams): [ApprovalState, (optionalParams?: OptionalApproveCallbackParams) => Promise<void>] {
   const { account, chainId } = useActiveWeb3React()
   const token = amountToApprove?.currency?.isToken ? amountToApprove.currency : undefined
   const currentAllowance = useTokenAllowance(token, account ?? undefined, spender)
   const pendingApproval = useHasPendingApproval(token?.address, spender)
 
+  // TODO: Nice to have, can be deleted
```

Comment:
yep, i would commented it out. I guess is useful, but in loal we have too much noise. 
Would it work just commenting, and u uncomment if you ever need it. I don't love dead code, but i like less to verbose logs